pkg/apis/provisioning/v0alpha1/types.go

```diff
@@ -295,6 +296,43 @@ type FileItem struct {
 	Author   string `json:"author,omitempty"`
 }
 
+// Information we can get just from the file listing
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ObjectList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []ObjectListItem `json:"items,omitempty"`
+}
+
+type ObjectListItem struct {
+	Path     string `json:"path"`
+	Group    string `json:"group"`
+	Resource string `json:"resource"`
+	Name     string `json:"name"` // the k8s identifier
+	Hash     string `json:"hash"`
```

Comment:
What if an object doesn't have a hash (i.e. from a non-versioned repo)? I think we should specify `omitempty` here.

Reply:
everything has a hash!  from local file system it is the [sha1](https://github.com/grafana/grafana/blob/grafana-git-ui-sync/pkg/registry/apis/provisioning/repository/local.go#L210), S3 has md5 etc

this is the value we can compare to see if anything has changed.